packages/cdk8s-plus/src/service.ts

```diff
@@ -168,14 +168,15 @@ export class Service extends Resource {
   /**
    * Associate a deployment to this service.
    *
-   * Requests will be routed to the port exposed by the first container in the
-   * deployment's pods. The deployment's `labelSelector` will be used to select
-   * pods.
+   * If not targetPort is specific in the portOptions, then requests will be routed
+   * to the port exposed by the first container in the deployment's pods. 
+   * The deployment's `labelSelector` will be used to select pods.
    *
    * @param deployment The deployment to expose
    * @param port The external port
+   * @param portOptions Optional settings for the port.
    */
-  public addDeployment(deployment: Deployment, port: number) {
+  public addDeployment(deployment: Deployment, port: number, portOptions: ServicePortOptions = {}) {
```

Comment:
```suggestion
  public addDeployment(deployment: Deployment, port: number, options: ServiceDeploymentOptions = {}) {
```

Reply:
@eladb I can't find ServiceDeploymentOptions and where these options get applied is a call to this.serve(port, {..options}) that only takes a ServicePortOptions.  I am changing the name, but may need a bit more of a pointer to understand the new type.

Reply:
I have renamed options.